internal/action/setup.go

```diff
@@ -0,0 +1,287 @@
+	out.Print(ctx, "🌟 Initializing a new password store ...")
+
+	if name := termio.DetectName(c.Context, c); name != "" {
+		ctx = ctxutil.WithUsername(ctx, name)
+	}
+	if email := termio.DetectEmail(c.Context, c); email != "" {
+		ctx = ctxutil.WithEmail(ctx, email)
+	}
+	// need to re-initialize the root store or it's already initialized
+	// and won't properly set up crypto according to our context.
+	s.Store = root.New(s.cfg)
+	inited, err := s.Store.IsInitialized(ctx)
+	if err != nil {
+		return ExitError(ExitUnknown, err, "Failed to initialized store: %s", err)
+	}
+	if inited {
+		out.Error(ctx, "⚠ Store is already initialized. Aborting.")
+		return nil
+	}
+
+	debug.Log("Starting Onboarding Wizard - remote: %s - team: %s - create: %t - name: %s - email: %s", remote, team, create, ctxutil.GetUsername(ctx), ctxutil.GetEmail(ctx))
+
+	crypto := s.getCryptoFor(ctx, team)
+	if crypto == nil {
+		return fmt.Errorf("can not continue without crypto")
+	}
+	debug.Log("Crypto Backend initialized as: %s", crypto.Name())
+
+	// check for existing GPG keypairs (private/secret keys). We need at least
+	// one useable key pair. If none exists try to create one
+	if !s.initHasUseablePrivateKeys(ctx, crypto) {
+		out.Print(ctx, "🔐 No useable cryptographic keys. Generating new key pair")
+		out.Print(ctx, "🕰 Key generation may take up to a few minutes")
+		if err := s.initGenerateIdentity(ctx, crypto, ctxutil.GetUsername(ctx), ctxutil.GetEmail(ctx)); err != nil {
+			return errors.Wrapf(err, "failed to create new private key")
+		}
+		out.Print(ctx, "🔐 Cryptographic keys validated")
+	}
+
+	debug.Log("Has useable private keys")
+
+	// if a git remote and a team name are given attempt unattended team setup
+	if remote != "" && team != "" {
+		if create {
+			return s.initCreateTeam(ctx, team, remote)
+		}
+		return s.initJoinTeam(ctx, team, remote)
+	}
+
+	// assume local setup by default, remotes can be added easily later
+	return s.initLocal(ctx)
+}
+
+func (s *Action) initGenerateIdentity(ctx context.Context, crypto backend.Crypto, name, email string) error {
+	out.Green(ctx, "🧪 Creating cryptographic key pair (%s) ...", crypto.Name())
+
+	out.Print(ctx, "🎩 Gathering information for the key pair ...")
+	name, err := termio.AskForString(ctx, "🚶 What is your name?", name)
+	if err != nil {
+		return err
+	}
+
+	email, err = termio.AskForString(ctx, "📧 What is your email?", email)
+	if err != nil {
+		return err
+	}
+
+	passphrase := xkcdgen.Random()
+	pwGenerated := true
+	if bv, err := termio.AskForBool(ctx, "⚠ Do you want to enter a passphrase? (otherwise we generate one for you)", false); err != nil && bv {
+		pwGenerated = false
+		sv, err := termio.AskForPassword(ctx, "✍ Please enter your passphrase")
+		if err != nil {
+			return errors.Wrapf(err, "Failed to read passphrase")
+		}
+		passphrase = sv
+	}
+
+	out.Yellow(ctx, "⏳ This can take a long time. If you get impatient see https://github.com/gopasspw/gopass/blob/master/docs/entropy.md")
```

Comment:
This shouldn't be the case since kernel version 5.6, since `/dev/random` now only blocks if the CPRNG hasn't been yet initialized and then it becomes non-blocking just like `/dev/urandom`.

Reply:
Kernel 5.6? You seem to be living in the future ;)

Let's keep this around for a little longer.